src/modalities/util.py

```diff
@@ -43,29 +47,60 @@ def parse_enum_by_name(name: str, enum_type: Type[Enum]) -> Enum:
+    the current date and time.
+    The experiment ID is then converted to a byte array (with maximum length of max_experiment_id_byte_length) and
+    broadcasted to all ranks. In the unlikely case of the experiment ID being too long, a ValueError is raised
+    and max_experment_id_byte_length must be increased. Each rank then decodes the byte array to the original
+    string representation and returns it. Having a globally synced experiment ID is mandatory for
+    saving files / checkpionts in a distributed training setup.
 
+    Args:
+        config_file_path (Path): Path to the configuration file.
+        hash_length (Optional[int], optional): Defines the char length of the commit hash. Defaults to 8.
+        max_experiment_id_byte_length (Optional[int]): Defines max byte length of the experiment_id
+            to be shared to other ranks. Defaults to 1024.
 
-def get_experiment_id_of_run(config_file_path: Path, hash_length: Optional[int] = 8) -> str:
-    """create experiment ID including the date and time for file save uniqueness
-    example: 2022-05-07__14-31-22_fdh1xaj2'
+    Returns:
+        str: The experiment ID.
     """
-    hash = hashlib.sha256(str(config_file_path).encode()).hexdigest()[:hash_length]
-    date_of_run = datetime.now().strftime("%Y-%m-%d__%H-%M-%S")
-    experiment_id = f"{date_of_run}_{hash}"
+
+    def get_experiment_id_from_config(config_file_path: Path, hash_length: Optional[int] = 8) -> str:
+        """Create experiment ID including the date and time for file save uniqueness
+        example: 2022-05-07__14-31-22_fdh1xaj2'
+        """
+        hash = hashlib.sha256(str(config_file_path).encode()).hexdigest()[:hash_length]
+        date_of_run = datetime.now().strftime("%Y-%m-%d__%H-%M-%S")
+        experiment_id = f"{date_of_run}_{hash}"
+        return experiment_id
+
+    rank = dist.get_rank()
+    if rank == 0:
+        # Generate a unique folder name
+        experimenet_id = get_experiment_id_from_config(config_file_path, hash_length)
+        experimenet_id_bytes = experimenet_id.encode("utf-8")
+        if len(experimenet_id_bytes) > max_experment_id_byte_length:
+            raise ValueError(
+                f"Experiment ID is too long: {len(experimenet_id_bytes)} bytes, "
+                f"max length is {max_experment_id_byte_length} bytes"
+            )
+        print(f"Rank 0 generated experiment_id: {experimenet_id}")
+    else:
+        experimenet_id_bytes = bytearray(max_experment_id_byte_length)  # Preallocate buffer for receiving
+
+    # Ensure all ranks have the same folder name
+    experment_id_tensor = torch.tensor(
+        list(experimenet_id_bytes) + [0] * (max_experment_id_byte_length - len(experimenet_id_bytes)), dtype=torch.uint8
+    ).cuda()
+    dist.broadcast(experment_id_tensor, src=0)
+
+    # Decode on all ranks
+    experiment_id = experment_id_tensor.cpu().numpy().tobytes().decode("utf-8").rstrip("\x00")
```

Comment:
If experiment_id_bytes endes with zero bytes (\x00, we strip parts of the id.
Broadcast also the lenght of the id?

Reply:
If a string "0" is encoded via utf-8, it has the byte representation \x30. 
\x00 is a non-printable control character, which can be used to indicate the end of a sequence. 
https://www.utf8-chartable.de/
https://en.wikipedia.org/wiki/Null_character